src/services/walletConnect/index.ts

```diff
@@ -48,7 +48,7 @@ export const walletConnectService = {
       address: accounts[0],
       source: 'walletConnect',
       name: 'WalletConnect',
-      signer,
+      signer: signer as WalletAccount['signer'], // TODO: improve - not type safe
```

Comment:
How would you improve this ? I personally don't agree with leaving TODO's around the code because, usually, they are never addressed. A better way, if you already know a way to improve this piece, but it is out of scope of this PR, is to create an issue and link it here.

Reply:
The problem currently is that there are some conflicting types between some libraries for the Signer type. This causes conflicts and TS erros when trying to use it. Now this is out of scope for this PR and requires a bit of time investment to fix it. Not sure about creating issues in zenhub for each small improvement in the code as it will cause a lot of noise. But will leave that up to you.

Reply:
imho, all small, medium and large improvements, should have a place to be tracked, and have a description/short explanation. Different thing is if we are going to fix that or not, but at least we know they exist. Once you have a big enough project, those issues might become more relevant and the priority to fix them may also arise.